pkg/cryptutil/certificates.go

```diff
@@ -219,6 +219,17 @@ func GenerateSelfSignedCertificate(domain string, configure ...func(*x509.Certif
 	return &cert, nil
 }
 
+// EncodeCertificate encodes a TLS certificate into PEM compatible byte slices.
+func EncodeCertificate(cert *tls.Certificate) (pemCertificateBytes, pemKeyBytes []byte) {
```

Comment:
Could we add a few unit tests to cover ` EncodeCertificate`?

1. Empty certificate
2. Nil `*tls.Certificate`
3. Malformed `tls.Certificate` `PrivateKey` ? Maybe not possible?
4. Others?

Reply:
Go does not have an easy way of generating certificates which aren't ed25519, ECDSA, or RSA. It's possible but I will have to import additional libraries and write functions to generate the certificates.